### PR TITLE
worker optimization

### PR DESCRIPTION
**Pull request by avikchaudhuri**
Change the worker management to be dynamic rather than static. Instead of dividing all files among the available number of processes, we now send chunks of work to each free process, and when a process finishes its chunk of work it gets another chunk. This leads to overall much better perf for a large number of files with non-uniform processing times.

### CODE REVIEW

File: src/Runner.js [hunk 1]
```diff
@@ -19,6 +19,7 @@ const fs = require('fs');
 const path = require('path');
 
 const availableCpus = require('os').cpus().length - 1;
+const CHUNK_SIZE = 50;
```
Comment (avikchaudhuri) [inline src/Runner.js]: Should I make this a config option?

File: src/Worker.js [hunk 2]
```diff
@@ -74,6 +78,9 @@ function trimStackTrace(trace) {
 function run(data) {
   var files = data.files;
   var options = data.options;
+  if (!files.length) {
+    finish();
+  } else
```
Comment (avikchaudhuri) [inline src/Worker.js]: Should format whatever comes next properly. Will fix.

File: src/Runner.js (hunk 1 above)
Comment (fkling) [inline src/Runner.js]: I'm fine with this being a constant for the time being.

**Comment by fkling** [comment]
Sweet, thank you! Will merge your PRs tomorrow.... maybe this also makes it a bit easier to solve #87.

**Comment by avikchaudhuri** [comment]
Not sure why the build failed, tests pass locally. Any way to kick off another build?

**Comment by cpojer** [comment]
btw. I'd recommend switching to node-worker-farm: https://github.com/rvagg/node-worker-farm

**Comment by fkling** [comment]
I just restarted it. Sometimes testing the Babel option fails, no idea why.

**Comment by avikchaudhuri** [comment]
Thanks! @cpojer Is it OK if we merge this PR now and we take on the work to use node-worker-farm later? I'm willing to investigate that option (looks really promising) but meanwhile I need some moar perf! :)

**Comment by fkling** [comment]
Released as v0.3.14. Thank you!

**Pull request merged**